src/Compilers/Core/Rebuild/BuildConstructor.cs

```diff
@@ -30,74 +30,64 @@ public class BuildConstructor
     {
         private readonly ILogger _logger;
 
+        // TODO: shouldn't need to pass a logger.
```

Comment:
>shouldn't need to pass a logger [](start = 17, length = 31)

Could we implement a simple `ILogger` that drops all log statements and use that here if `logger == null`?

Reply:
We could but the goal is to just delete this field from the type entirely. I came close to making that change in this PR but decided to hold back and save it for another.